### PR TITLE
Add more graceful handling for go type assertions in `runtime/sema`

### PR DESCRIPTION
**Pull request by dsainati1**
## Description

This adds more graceful go handling for some type assertions in the `runtime/sema` directory, which previously were unhandled. This should not result in any behavioral differences. 

______

- [X] Targeted PR against `master` branch
- [X] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [X] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [X] Updated relevant documentation 
- [X] Re-reviewed `Files changed` in the Github PR explorer
- [X] Added appropriate labels

### CODE REVIEW

File: languageserver/server/server.go [hunk 1]
```diff
@@ -753,12 +753,18 @@ func (s *Server) DocumentHighlight(
 
 	position := conversion.ProtocolToSemaPosition(params.Position)
 	occurrences := checker.Occurrences.FindAll(position)
+	if occurrences == nil {
+		return nil, nil
+	}
```
Comment (SupunS) [inline languageserver/server/server.go:758]: If the result is nil, then the below condition should get executed. i.e:`len()` of a nil array is also zero.

It is safe to skip the nil check here and below, as `len()` and `range` expressions are nil-safe.

File: runtime/sema/member_accesses.go [hunk 2]
```diff
@@ -57,15 +57,22 @@ func (m *MemberAccesses) Find(pos Position) *MemberAccess {
...
 }
 
 func (m *MemberAccesses) All() []MemberAccess {
 	values := m.tree.Values()
 	accesses := make([]MemberAccess, len(values))
 	for i, value := range values {
-		accesses[i] = value.(MemberAccess)
+		access, ok := value.(MemberAccess)
+		if !ok {
+			return nil
```
Comment (SupunS) [inline runtime/sema/member_accesses.go:73]: would `continue` be a better option? I'm ok with both.

**Review by SupunS — APPROVED** [review]
Nice! 👌

File: runtime/sema/member_accesses.go (hunk 2 above)
Comment (dsainati1) [reply runtime/sema/member_accesses.go:73]: I think this depends on what we'd like this function to do. If one of the elements of `values` is not a `MemberAccess` even though this function is meant to be called on a `MemberAccesses` collection, that feels like a significant enough issue that we'd like to fail the entire operation and return `nil`. However, if you feel like we should just recover as much as possible and return those values that are valid then we can `continue`.

Comment (SupunS) [reply runtime/sema/member_accesses.go:73]: I felt returning `nil` would be somewhat closer to returning an empty array (the result of this is more likely to be looped over the `range`). Maybe if we want to completely fail the operation then panicking with a dedicated error would be even better I guess.
My initial thought was, if we are silently failing, we could make it to be resilient as much as possible.

**Pull request merged**